waspc/data/Generator/templates/sdk/wasp/universal/types.ts

```diff
@@ -29,3 +29,7 @@ export type _Awaited<T> = T extends Promise<infer V>
 // consideration.
 export type _ReturnType<T extends (...args: never[]) => unknown> = 
   T extends (...args: never[]) => infer R ? R : never
+
+
+// Returns elements of an array except the first one.
+export type Tail<T extends unknown[]> = T extends [unknown, ...infer R] ? R : never;
```

Comment:
The result would be an empty set e.g. `never` which seems appropriate. For every array with at least one element -> return an array with one less element. For an empty array that transformation is not possible. 

Since I suspect we'll always use this helper with a known list e.g. a list of params of a function we know the signature for, this seems okay? We'll know when the `Tail` helper can be used.

Reply:
Right, I'm ok with it.

I'll just point out, in case it didn't occur to you, that you could easily enforce the passed arrays to be non-empty and, since everything happens in compile time, I assume dealing with non-empty lists won't be as annoying as it is runtime:
```typescript
export type Tail<T extends [unknown, ...unknown[]]> = T extends [unknown, ...infer R] ? R : never;
```
Feel free to do what you want and resolve.

Reply:
Nice tip, didn't really think I could do this. I'll implement this 👍